### ISSUE TITLE
`IAsyncEnumerable` generates an infinite sequence 🐛 

### ISSUE THREAD

**Issue by Kralizek** [AutoFixture/AutoFixture#1446]
### Describe the Bug
I have a service accepting a dependency that returns a `IAsyncEnumerable<T>` and consumes it.

When I use AutoFixture and FakeItEasy, the call produces an infinite sequence.

Support for `IAsyncEnumerable<T>` was added here: https://github.com/AutoFixture/AutoFixture/pull/1323

I suspect the problem is caused by registering FakeItEasy as a customization.

### Scenario

Here is a stripped-down example.

```csharp
public class Service(IDependency dependency)
{
	public async Task<String> GetString()
	{
		var sb = new StringBuilder();
		
		await foreach (var s in dependency.GetStrings())
		{
			sb.Append(s);
		}
		
		return sb.ToString();
	}
}

public interface IDependency
{
	IAsyncEnumerable<string> GetStrings();
}
```

The code below will never return.

```csharp
void Main()
{
	var fixture = new Fixture();

	fixture.Customize(new AutoFakeItEasyCustomization { ConfigureMembers = true, GenerateDelegates = true });
	var svc = fixture.Create<Service>();

	svc.GetString().Dump();
}
```

### Expected Behavior

Given how `IAsyncEnumerable<T>` is configured after #1323, it should return three strings.

The code works when using FakeItEasy directly

```csharp
void Main()
{
	var fixture = new Fixture();

	var dep = A.Fake<IDependency>();
	A.CallTo(() => dep.GetStrings()).Returns(new string[] { "hello", "world"}.ToAsyncEnumerable());
	var svc = new Service(dep);
	
	svc.GetString().Dump();
}
```

### Tasks

- [x] Updated all AutoFixture and related libraries
- [ ] Confirmed in support forums if behavior is expected

Thanks!

**Comment by aivascu** [comment]
@Kralizek I assume you've checked the issue in v4 version doesn't have the `AsyncEnumerableRelay`, and there are no plans to add it retroactively. As a reminder #1323 has been merged into the `release` branch version and not `master`. )
The issue does not reproduce on v5 preview 8

As a workaround you could add a customization to relay the request to a regular collection, or build your own engine parts for the fixture. I'm trying to make some time in my schedule to push v5 in the coming weeks, so hopefully you'll not need it for long. 🤞
```cs
fixture.Customize<IAsyncEnumerable<string>>(c => c.FromFactory((string[] x) => x.ToAsyncEnumerable()));
```

**Comment by Kralizek** [comment]
Thanks, I realized the problem was elsewhere. And, honestly, I totally forgot that #1323 was merged in 5.0 and not 4.0. My bad. We can close this one.
